Automatic Desired Area Function for XYDelaunay

miaoyinb opened the pull request:
closes #27433

Pending https://github.com/libMesh/libmesh/pull/3813

<!--
If this PR is associated with an issue be sure to reference it
by including "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).

If this PR implements an enhancement that does not have an existing issue, please add the following:
-->

## Reason
<!--Why do you need this feature or what is the enhancement?-->

## Design
<!--A concise description (design) of the enhancement.-->

## Impact
<!--Will the enhancement change existing APIs or add something new?-->


<!--
If this PR implements a bug fix, please create an issue for the bug and reference the issue.
-->

miaoyinb committed:
Automatic Desired Area Function for XYDelaunay

closes #27433

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2188449/) on e284409 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/27434/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/27434/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 3b9daf46a4c95ca51f405abe6d6295df54c7e407`

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit 0aa651ce180a197006f8f34b7a20cd9a6e30f305
+Subproject commit 79f8bcf4c31721323622414ef59d33f3db7ac7c5
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2188449/) on e284409 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml
  conda/libmesh/meta.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2188496/) on 33f3189 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml
  conda/libmesh/meta.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2188496/) on 33f3189 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml
  conda/libmesh/meta.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2206322/) on 306df28 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml
  conda/libmesh/meta.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2206318/) on 306df28 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/27434/site)

This comment will be updated on new commits.

miaoyinb commented:
@GiudGiud @roystgnr 
I was wondering what is the routine to do a PR that needs a libMesh update. Any suggestion would be appreciated.

Meanwhile, I will try to fix the current error, I think it is related to the randomness in triangulation when using different numbers of cores.

GiudGiud commented:
Usually we do libmesh updates in a separate PR that updates the submodule. There's nothing too special about it except you need to tell Logan or Jason about it so they look at / add deployment items in the PR

miaoyinb committed:
use csvdiff instead of exodiff to accomodate randomness

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2208768/) on 7bf5d62 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml
  conda/libmesh/meta.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Framework 2](https://civet.inl.gov/job/2208789/) on 7bf5d62 : invalidated by @miaoyinb

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2208808/) on 7bf5d62 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>14e79c</th>
      <th colspan=4>#27434 7bf5d6</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.11%</td>
      <td>85.11%</td>
      <td>+0.00%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>103108</td>
      <td>103133</td>
      <td>+25</td>
      <td>29</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>18044</td>
      <td>18044</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/27434/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/27434/coverage/framework)


## Modules coverage

### Reactor

<table>
  <thead>
    <tr>
      <th></th>
      <th>14e79c</th>
      <th colspan=4>#27434 7bf5d6</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>94.09%</td>
      <td>94.12%</td>
      <td>+0.03%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>6158</td>
      <td>6191</td>
      <td>+33</td>
      <td>33</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>387</td>
      <td>387</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/27434/coverage_diff/reactor)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/27434/coverage/reactor)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/27434/coverage/xfem)

</details>



This comment will be updated on new commits.

miaoyinb commented:
@GiudGiud I think all the tests are good except for those relying on conda. 

@milljm @loganharbour I would appreciate it a lot if you could help with the libmesh submodule update. Thanks in advance.

GiudGiud commented on framework/doc/content/source/meshgenerators/XYDelaunayGenerator.md:
```diff
@@ -7,9 +7,9 @@ generated output mesh, as well as to specify any Steiner points
-generators.
+function of triangle center location that can be specified or
+automatically generated).  Additional interior boundary "holes" can
+be generated from inputs from other specified "holes" mesh generators.
```
```suggestion
be meshed by the mesh generators specified in the [!param](//Mesh/XYDelaunayGenerator/holes) vector parameter.
```

GiudGiud commented on framework/src/meshgenerators/XYDelaunayGenerator.C:
```diff
@@ -76,6 +76,27 @@ XYDelaunayGenerator::validParams()
       std::string(),
       "Desired area as a function of x,y; omit to skip non-uniform refinement");
+
+  params.addParam<bool>("use_auto_area_func", false, "Use the automatic area function.");
```
```suggestion
  params.addParam<bool>("use_auto_area_func", false, "Use the automatic area function for the triangle meshing region.");
```

GiudGiud commented on framework/src/meshgenerators/XYDelaunayGenerator.C:
```diff
@@ -85,6 +106,10 @@ XYDelaunayGenerator::validParams()
 
+  params.addParamNamesToGroup(
+      "use_auto_area_func auto_area_func_default_size auto_area_func_default_size_dist",
+      "Automatic Area Function");
```
```suggestion
      "Automatic triangle meshing area control");
```
I would add the other parameters for that region in another parameter group

GiudGiud commented on modules/reactor/src/meshgenerators/FlexiblePatternGenerator.C:
```diff
@@ -84,6 +84,27 @@ FlexiblePatternGenerator::validParams()
       std::string(),
       "Desired area as a function of x,y; omit to skip non-uniform refinement");
+
+  params.addParam<bool>("use_auto_area_func", false, "Use the automatic area function.");
```
```suggestion
  params.addParam<bool>("use_auto_area_func", false, "Use the automatic area function for triangle-meshing in the background.");
```

GiudGiud commented on modules/reactor/src/meshgenerators/PeripheralTriangleMeshGenerator.C:
```diff
@@ -37,13 +37,38 @@ PeripheralTriangleMeshGenerator::validParams()
+  params.addParamNamesToGroup("desired_area desired_area_func use_auto_area_func "
+                              "auto_area_func_default_size auto_area_func_default_size_dist "
+                              "auto_area_function_num_points auto_area_function_power",
+                              "Area Limit Delaunay");
```
```suggestion
                              "Peripheral Area Delaunay");
```

GiudGiud commented on modules/reactor/src/meshgenerators/PeripheralTriangleMeshGenerator.C:
```diff
@@ -37,13 +37,38 @@ PeripheralTriangleMeshGenerator::validParams()
       std::string(),
       "Desired area as a function of x,y; omit to skip non-uniform refinement");
+
+  params.addParam<bool>("use_auto_area_func", false, "Use the automatic area function.");
```
```suggestion
  params.addParam<bool>("use_auto_area_func", false, "Use the automatic area function in the peripheral region.");
```

GiudGiud commented on test/tests/meshgenerators/xy_delaunay_generator/tests:
```diff
@@ -112,5 +123,13 @@
+      cli_args = '--mesh-only err.e
+                  Mesh/triang/use_auto_area_func="true"'
+      expect_err = "Only one of the three methods"
+      detail = "if more than one element area limit methods are enabled"
```
```suggestion
      detail = "if more than one element area limiting methods are specified by the user"
```

GiudGiud commented on modules/reactor/doc/content/source/meshgenerators/PeripheralTriangleMeshGenerator.md:
```diff
@@ -6,7 +6,7 @@
 The `PeripheralTriangleMeshGenerator` utilizes the general purpose [`XYDelaunayGenerator`](XYDelaunayGenerator.md) triangulation mesh generator and the [`PolyLineMeshGenerator`](PolyLineMeshGenerator.md) to create a circular periphery region around a reactor core. The outer circle properties are specified with the [!param](/Mesh/PeripheralTriangleMeshGenerator/peripheral_ring_radius) and [!param] ...
 
-Optionally, triangle refinement parameters can be used to improve the quality of the triangles in the periphery region. These are specified with either [!param](/Mesh/PeripheralTriangleMeshGenerator/desired_area), which specifies a single desired maximum triangle area, and [!param](/Mesh/PeripheralTriangleMeshGenerator/desired_area_func), which specifies the desired maximum triangle area as a fun ...
+Optionally, triangle refinement parameters can be used to improve the quality of the triangles in the periphery region. These are specified with either [!param](/Mesh/PeripheralTriangleMeshGenerator/desired_area), which specifies a single desired maximum triangle area, and [!param](/Mesh/PeripheralTriangleMeshGenerator/desired_area_func), which specifies the desired maximum triangle area as a fun ...
```
```suggestion
Optionally, triangle refinement parameters can be used to improve the quality of the triangles in the periphery region. These are specified with either the [!param](/Mesh/PeripheralTriangleMeshGenerator/desired_area) parameter, which specifies a single desired maximum triangle area, or the [!param](/Mesh/PeripheralTriangleMeshGenerator/desired_area_func) parameter, which specifies the desired maximum triangle area as a function of the spatial coordinates x and y. Alternatively, this type of refinement can be done following an automatically generated area function. This feature is enabled by [!param](/Mesh/XYDelaunayGenerator/use_auto_area_func), while the related parameters include [!param](/Mesh/XYDelaunayGenerator/auto_area_func_default_size), [!param](/Mesh/XYDelaunayGenerator/auto_area_func_default_size_dist), [!param](/Mesh/XYDelaunayGenerator/auto_area_function_num_points), and [!param](/Mesh/XYDelaunayGenerator/auto_area_function_power). See the full documentation for [`XYDelaunayGenerator`](XYDelaunayGenerator.md) for more details on the area refinement options.
```

GiudGiud reviewed (commented):
looks good!
CIVET's message tells you which files to modify if you want to try to get these tests green. We will have to have the deployment team review it regardless though so feel free to leave that to them if you prefer

miaoyinb committed:
Apply suggestions from code review

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

Update framework/doc/content/source/meshgenerators/XYDelaunayGenerator.md

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

minor typo in doc

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2211214/) on 6f91a24 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/27434/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/27434/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b3efaa2fb20af48d2c07a376c6c0f653ae8a66db`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2211214/) on 6f91a24 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml
  conda/libmesh/meta.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2211261/) on 957263d wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml
  conda/libmesh/meta.yaml

The libmesh submodule or configuration was changed but the conda build config was not

GiudGiud commented on framework/doc/content/source/meshgenerators/XYDelaunayGenerator.md:
```diff
@@ -55,6 +55,31 @@ simple Laplacian smoother; this can have unwanted side-effects on
 
+The elements generated by `XYDelaunayGenerator` can be refined according
+to a given maximum element area. This element area limit can be provided
+using one the the following three approaches:
```
```suggestion
using one of the following three approaches:
```

GiudGiud commented on framework/src/meshgenerators/XYDelaunayGenerator.C:
```diff
@@ -101,9 +128,21 @@ XYDelaunayGenerator::XYDelaunayGenerator(const InputParameters & parameters)
 {
+  if ((_desired_area > 0.0 && !_desired_area_func.empty()) ||
+      (_desired_area > 0.0 && _use_auto_area_func) ||
+      (!_desired_area_func.empty() && _use_auto_area_func))
```
We ll want to check for all the automatic area parameters here instead. Otherwise they can be specified by the user then ignored because another method is in use

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2211678/) on d40171e wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml
  conda/libmesh/meta.yaml

The libmesh submodule or configuration was changed but the conda build config was not

miaoyinb committed:
check auto area function parameters

miaoyinb committed:
update libmesh submodule

miaoyinb commented:
> looks good! CIVET's message tells you which files to modify if you want to try to get these tests green. We will have to have the deployment team review it regardless though so feel free to leave that to them if you prefer

I will try it just out of curiosity.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2211799/) on d55a70f wanted to post the following:

A change of the following file(s) triggered this check:

  conda/libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2211799/) on d55a70f wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2211799/) on d55a70f wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented:
@miaoyinb well done on the Conda side of things! There exists one more thing required: updating the `versioner_hashes.yaml` file in `moose/scripts/tests`.

It is generally the very last thing you want to do. And because it looks like some things are still failing, you will want to hold off on the following instructions until then.

The version hash file allows us to keep a solid record tree on all the updates occurring with our Conda packages (as well as Apptainer definition files).

> [!TIP]
> The following instructions are also available within and at the beginning of the following file: `moose/scripts/tests/versioner_hashes.yaml`


1. get the current HASH of your completed PR:
    ```bash
    git rev-parse HEAD
    ```
    record this information for later use.
2. open the file `moose/scripts/tests/versioner_hashes.yaml` and append the above output while also providing the MOOSE PR responsible for making these changes using the following syntax:
    ```pre
    <HASH>: #27433
    ```
3. run our `versioner.py` script located in `moose/scripts` on all libraries we keep tabs on (libMesh being one of them, thus why you need to do these steps):
    ```bash
    ./versioner.py mpich
    ./versioner.py petsc
    ./versioner.py libmesh
    ./versioner.py moose-dev
    ./versioner.py wasp
    ```
5. add each key and assign the corresponding value gathered from the above output, using the following syntax:
    ```pre
    <HASH>: #27433
      mpich: <hash>
      petsc: <hash>
      libmesh: <hash>
      moose-dev: <hash>
      wasp: <hash>
    ```
    *because you are only updating libMesh, you should see no changes for mpich, petsc, and wasp entries from the previous HASH: #PR entry*
6. `git add; git commit; git push` one last time.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2212380/) on b1f2f8d wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2212380/) on b1f2f8d wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2213910/) on 125cf9a wanted to post the following:

A change of the following file(s) triggered this check:

  conda/libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2213910/) on 125cf9a wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2213910/) on 125cf9a wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Test debug](https://civet.inl.gov/job/2213948/) on 125cf9a : invalidated by @miaoyinb

moosebuild commented:
Job [Public app tests](https://civet.inl.gov/job/2213940/) on 125cf9a : invalidated by @miaoyinb

moosebuild commented:
Job [Modules parallel](https://civet.inl.gov/job/2213936/) on 125cf9a : invalidated by @miaoyinb

moosebuild commented:
Job [Framework 1](https://civet.inl.gov/job/2213930/) on 125cf9a : invalidated by @miaoyinb

moosebuild commented:
Job [No optional](https://civet.inl.gov/job/2213946/) on 125cf9a : invalidated by @miaoyinb

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2213950/) on 125cf9a : invalidated by @miaoyinb

miaoyinb commented:
I do not think the "controlled app tests" failure is related, but I cannot see or invalidate it.

GiudGiud commented:
@milljm  can you please sign off on the libmesh update items?
this is otherwise ready to merge

miaoyinb commented:
@milljm @GiudGiud Let me know if any additional modifications are needed. Thanks.

GiudGiud commented:
@miaoyinb looks like somebody updated libmesh elsewhere?

GiudGiud commented:
Squashed minor commits and removed libmesh update. Will merge if libmesh submodule is far enough forward now

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2228764/) on 44dcbe9 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml
  conda/libmesh/meta.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2228823/) on 4a7554c wanted to post the following:

A change of the following file(s) triggered this check:

  conda/libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2228823/) on 4a7554c wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2228823/) on 4a7554c wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2228989/) on 8ff2fb4 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2228989/) on 8ff2fb4 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2228989/) on 8ff2fb4 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

miaoyinb commented:
@GiudGiud I am on travel this week and just landed. Will work on this. Thanks.

miaoyinb commented:
I think `moose-dev` and `wasp` have been updated, but `moose-libmesh` seems not...

miaoyinb commented:
@GiudGiud it seems you already worked on it. Thanks. Do I need to do anything else?

GiudGiud commented:
should be fine once I get those details right

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2229180/) on 0fa90f7 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2229180/) on 0fa90f7 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2229180/) on 0fa90f7 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2229808/) on 0bee359 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2229808/) on 0bee359 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2229808/) on 0bee359 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

miaoyinb committed:
version hashes

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2229902/) on 1af0117 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2229902/) on 1af0117 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented:
The Conda stuff looks good to me! Well Done.

I need to allow others that know more about libMesh to approve the PR.

GiudGiud reviewed (approved):
@roystgnr signed off on updating in a pm

GiudGiud commented:
Thanks for the contribution!
I think moving forward we'll want to keep the libmesh updates in separate PRs. It will be easier to bring all the relevant expertise together

loganharbour commented:
This broke next https://civet.inl.gov/event/180188/

miaoyinb commented:
7500cf899b0b838167fae6f5a1eb5c7a63dc2f09

This is still caused by randomness in triangulation. I did a comprehensive test and this commit should fix it.
What's the best way to fix this? A new PR?

GiudGiud commented:
new PR is the way
